design/baremetal-operator/node-specific-kernel-args.md

```diff
@@ -0,0 +1,166 @@
+<!--
+ This work is licensed under a Creative Commons Attribution 3.0
+ Unported License.
+
+ http://creativecommons.org/licenses/by/3.0/legalcode
+-->
+
+# pre-provisioning kernel argument support
+
+## Status
+
+planned
+
+## Summary
+
+This document proposes a new BareMetalHost spec attribute intended for
+configuring the pre-provisioning image's (usually IPA) kernel arguments.
+
+## Motivation
+
+There are situations where a user would need node specific kernel arguments
+for the pre-provisioning image. The reason could be the use of a mixed set of
+hardware, troubleshooting provisioning or boot issues, selectively configuring
+custom logic e.g. enabling IPA plugins or other services embedded in the
+pre-provisioning image.
+
+### Goals
+
+1. Implement a string array type optional spec field named
+   `preProvisionKernelArgs`
```

Comment:
Please keep spelling and word choice consistent: `preprovisioningKernelParams`

Reply:
Actually, it's called `ExtraKernelParams` in PPI. There is a reason for that: these options are appended to the defaults in ironic-image. I'd prefer to keep it this way for the new feature, otherwise it won't be usable for anyone with ironic-image-level customizations (like we have to provide something for CoreOS/Ignition).